lib/cart.rb

```diff
@@ -0,0 +1,51 @@
+require_relative "menu_item"
+require_relative "menu"
+require_relative "order_item"
+
+class Cart
+  attr_accessor :cart
+  attr_accessor :total_cost
+
+  def initialize
+    @cart = []
+    @total_cost = 0
+  end
+
+  @@instance = Cart.new
+
+  def self.instance
+    return @@instance
+  end
+
+  def add_item_to_cart(order_item)
+    @cart.push(order_item)
+    cost = order_item.cost.to_i
+    @total_cost += cost
+  end
+
+
+  def to_s
+    puts "You ordered:"
+    cart.each do |order_item|
+      puts "#{order_item.name}: #{order_item.quantity} x #{order_item.unit_price} #{order_item.currency} = #{order_item.cost} #{order_item.currency}"
+    end
+    puts "GRAND TOTAL: #{@total_cost}"
+  end
+
+  private_class_method :new
+end
+
+if __FILE__ == $0
```

Comment:
I wanted to test out the __FILE__ == $0 thing that was in the tutorial you had us do on Thursday: https://www.ruby-lang.org/en/documentation/quickstart/4/.  I found it pretty neat and useful to iterate on my cart to get it to 'work' (in quotes because who knows whether it really works is still highly uncertain, even doubtful).  But this is a great learning experience, that's for sure!